api/capi/include/nnstreamer-capi-private.h

```diff
@@ -157,6 +157,14 @@ typedef struct _ml_pipeline_element {
   GMutex lock; /**< Lock for internal values */
 } ml_pipeline_element;
 
+/**
+ * @brief Internal data structure for the pipeline state callback.
+ */
+typedef struct {
+  ml_pipeline_state_cb cb; /**< Callback to notify the change of pipeline state */
+  void *user_data; /**< The user data passed when calling the state change callback */
+} pipeline_state_cb_s;
+
```

Comment:
What's the benefit of this refactoring?

Reply:
Now I am considering another callback for the resource control in Pipeline struct (e.g., tizen mmfw). This is for the readability.

Reply:
If we may add another callback, then, it's ok.